### PR TITLE
protocol service Stop() fix

### PR DESCRIPTION
**Pull request by GuanLuo**
Add @ryanolson as part of the fix is on nvRPC

### CODE REVIEW

File: src/core/http_server.cc [hunk 1]
```diff
@@ -129,15 +135,25 @@ tensorflow::Status
 HTTPServerImpl::Stop()
```
Comment (deadeyegoodwin) [inline src/core/http_server.cc]: So does Stop() now actually cause the server to stop?

File: src/nvrpc/Server.cc [hunk 2]
```diff
@@ -77,6 +77,7 @@ Server::AsyncRun()
 void
 Server::Shutdown()
 {
+  m_Server->Shutdown();
```
Comment (deadeyegoodwin) [inline src/nvrpc/Server.cc]: Does this cause Stop() to stop the GRPC service?

Comment (GuanLuo) [reply src/nvrpc/Server.cc]: Stop() now will actually stop the service and return

Stop() calls this function, and then procedure will follow the shutdown process suggested in [GRPC doc](https://grpc.io/grpc/cpp/classgrpc_1_1_server_builder.html#a960d55977e1aef56fd7b582037a01bbd)

Comment (deadeyegoodwin) [reply src/nvrpc/Server.cc]: So if --allow-poll-model-repository the server will just exit right away? Also I think on shutdown the main thread is going to exit before Close() is complete.

Comment (GuanLuo) [reply src/nvrpc/Server.cc]: Yes. And I think you are right, the main thread will exit before Close() is complete in this case. Do we have to call Close() on separate thread? If not, we can only set server state to SERVER_EXITING on receiving signal and call Close() after Wait() return.

Comment (deadeyegoodwin) [reply src/nvrpc/Server.cc]: For the library work I'm redoing the main.cc for trtserver so I will fix it as part of that change. I'll just hold off on merging this until my part of the change is ready.

**Comment by deadeyegoodwin** [comment]
Subsumed by https://github.com/NVIDIA/tensorrt-inference-server/pull/159

**Pull request closed without merging**